### PR TITLE
[MU4] Added key navigation from top tools

### PR DESCRIPTION
**Pull request by igorkorsukov**
incuding https://github.com/musescore/MuseScore/pull/7908

https://user-images.githubusercontent.com/3818029/113883465-76f0b900-97be-11eb-9441-64b0c1e2c1e5.mp4

### CODE REVIEW

File: src/appshell/qml/DevTools/KeyNav/KeyNavExample.qml [hunk 1]
```diff
@@ -1,62 +1,155 @@
...
+        id: infoPanel
 
-            sectionName: "sect 2"
-            sectionOrder: 2
-            subsecCount: 3
+        anchors.left: parent.left
+        anchors.right: parent.right
+        height: 64
+
+        Text {
```
Comment (RomanPudashkin) [inline src/appshell/qml/DevTools/KeyNav/KeyNavExample.qml:19]: Please, use StyledTextLabel here (even in devtools)

Comment (igorkorsukov) [reply src/appshell/qml/DevTools/KeyNav/KeyNavExample.qml]: fixed

File: src/framework/ui/ikeynavigation.h [hunk 2]
```diff
@@ -16,38 +16,62 @@
...
-    virtual ~IKeyNavigationSection() = default;
+    virtual ~IKeyNavigationControl() = default;
 
-    virtual QString name() const = 0;
-    virtual int order() const = 0;
-    virtual bool enabled() const = 0;
-    virtual bool active() const = 0;
-    virtual void setActive(bool arg) = 0;
+    virtual void trigger() = 0;
+    virtual async::Channel<IKeyNavigationControl*> forceActiveRequested() const = 0;
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply src/framework/ui/ikeynavigation.h:55]: Not this case.
In this case, instances of `IKeyNavigationControl` this is Qml objects, so impossible to use shared_ptr

File: src/framework/ui/internal/keynavigationcontroller.cpp [hunk 3]
```diff
@@ -304,21 +372,130 @@ void KeyNavigationController::prevSubSection()
...
+        if (first) {
+            first->setActive(true);
+        }
+        return;
+    }
+
+    nextControl->setActive(true);
+}
+
+void KeyNavigationController::prevControl()
```
Comment (RomanPudashkin) [inline src/framework/ui/internal/keynavigationcontroller.cpp:426]: I think a better name for this method is goToPrevControl (if I understood correctly)

File: src/framework/ui/internal/keynavigationcontroller.cpp [hunk 4]
```diff
@@ -304,21 +372,130 @@ void KeyNavigationController::prevSubSection()
...
+        }
+        return;
+    }
+
+    activateSubSection(prevSubSec);
+}
+
+void KeyNavigationController::nextControl()
+{
+    LOGI() << "====";
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply src/framework/ui/internal/keynavigationcontroller.cpp:396]: Not now, I will remove all debug when completed most tasks about key navigation.
At the moment I need this log

File: src/framework/ui/view/keynavigationcontrol.h [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace mu::ui {
+class KeyNavigationControl : public AbstractKeyNavigation, public IKeyNavigationControl, public async::Asyncable
+{
+    Q_OBJECT
+    Q_PROPERTY(KeyNavigationSubSection * subsection READ subsection WRITE setSubSection NOTIFY subsectionChanged)
+
+public:
+    explicit KeyNavigationControl(QObject* parent = nullptr);
+    ~KeyNavigationControl();
```
Comment (RomanPudashkin) [inline src/framework/ui/view/keynavigationcontrol.h:37]: You missed override here

File: src/framework/uicomponents/qml/MuseScore/UiComponents/GradientTabButton.qml [hunk 6]
```diff
@@ -32,13 +53,16 @@ RadioDelegate {
...
+
+            border.color: ui.theme.focusColor
+            border.width: keynavCtrl.active ? 2 : 0
             radius: 2
         }
 
         Item {
             id: backgroundGradientRect
             anchors.fill: parent
+            anchors.margins: 2
```
Comment (RomanPudashkin) [inline src/framework/uicomponents/qml/MuseScore/UiComponents/GradientTabButton.qml:65]: Please explain why you need this margin

File: src/framework/ui/uitypes.h [hunk 7]
```diff
@@ -60,6 +60,7 @@ enum ThemeStyleKey
     FONT_PRIMARY_COLOR,
     FONT_SECONDARY_COLOR,
     LINK_COLOR,
+    FOCUS_COLOR,
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply src/framework/ui/uitypes.h:63]: Added, please review https://github.com/musescore/MuseScore/pull/7909/files#diff-0a2fe9c9322a8317cb387aa9f247cacc31c90bcc4bb6983e1876c2db91269c3c

File: src/framework/ui/internal/keynavigationcontroller.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/framework/ui/internal/keynavigationcontroller.cpp]: renamed

File: src/framework/ui/view/keynavigationcontrol.h (hunk 5 above)
Comment (igorkorsukov) [reply src/framework/ui/view/keynavigationcontrol.h]: added

File: src/framework/uicomponents/qml/MuseScore/UiComponents/GradientTabButton.qml (hunk 6 above)
Comment (igorkorsukov) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/GradientTabButton.qml]: added NOTE

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**